### ISSUE TITLE
Arbogen generates objects of the wrong size

### ISSUE THREAD

**Issue by Kerl13** [fredokun/arbogen#27]
```shell
./bin/arbogen.native -seed 42424242 -min 10 -max 20 examples/binary.spec

                A      ...:'....:'':...':......
                R    :''   ._   .  `.  \   ,   '':
                B    ':  .   " .|    \  `>/   _.-':
                O   .:'  .`'.   `-.  '. /'  ,..  .:
                G  :'        `.    `\| \./   ' :
                E  :. ,,-'''''  "-.   |   | ....:
                N   '.      ..'''  `\ :   |
                ''''''''       \'   |
                *fast* uniform random    |  =|
                tree generator    |   |
                |-  |
                '''''''''''''''''''''''''''''''''''''''
                (C) F. Peschanski et al. under the GPL

Loading grammar file: examples/binary.spec
==> Grammar file loaded
Generating tree
Random number generator used  is ocaml
==> Tree generated with 23 nodes
[[[[],[[],[]]],[[[[],[]],[]],[]]],[]]
```

**Comment by Kerl13** [comment]
and of the wrong type
[tree.pdf](https://github.com/fredokun/arbogen/files/3122205/tree.pdf)

**Comment by Kerl13** [comment]
It turns out the output was correct:

- The displayed size (in the example: 23) took the leaves into account which is incorrect since they have a 0 weigth (fixed in 53debad93a12bc3016f3b23de51c4193bfb46518)

- The tree is actually binary but the generated `.dot` file is confusing since it displays leaves using two nodes
